src/main/webapp/scripts/my-trips.js

```diff
@@ -222,6 +298,36 @@ function parseAndRenderHotelResults(json) {
   }
 }
 
+/**
+ * Implementation of the Haversine formula, recommended by NASA to calculate
+ * distances between two coordinate pairs based on Latitude and Longitude
+ * (source: https://andrew.hedges.name/experiments/haversine/)
+ * @param {Object} p1 a coordinate pair with fields lat and lng
+ * @param {Object} p2 a coordinate pair with fields lat and lng
+ * @returns {number} the distance between the two in km
+ */
+function distanceBetween(p1, p2) {
+  // Earth mean radius - 6371 km by Google
+  const lngDelta = degToRad(p2.lng - p1.lng);
+  const latDelta = degToRad(p2.lat - p1.lat);
+  const a =
+    Math.sin(latDelta / 2) ** 2 +
+    Math.cos(degToRad(p1.lat)) *
+      Math.cos(degToRad(p2.lat)) *
+      Math.sin(lngDelta / 2) ** 2;
```

Comment:
nit: these two lines are tabbed one extra

Reply:
This is intentional. I think Prettier prefers this for long arithmetic operations.